### DISCUSSION TITLE
mosh-server 1.4.0 homebrew version 1.4.-3-gc16108f-homebrew failure

### DISCUSSION THREAD

**Discussion by tthkbw** [blinksh/blink#1817] [Support] [answered]
blink version v16.1.11.798 (App Store) Jul 21 2023

From my 2019 iPad Pro running iPadOS 16.6 I can no longer connect to my MacBook Air M2. Blink reports "Nothing received from server on UDP port 60002" and then:

```
blink> mosh macair
Connected to fe80::48d2:bdff:fe96:48ee%anpi0

mosh did not make a successful connection to 192.168.11.36:60002.
Please verify that UDP port 60002 is not firewalled and can reach the server.

(By default, mosh uses a UDP port between 60000 and 61000. The -p option
                                                                        selects a specific UDP port number.)

Mosh session finished!
blink> 
```
The firewall on the MacBook is disabled.

I upgraded homebrew on the MacBook and it is now "build mosh-1.4.0-3-gc16108f-Homebrew". I am running macOS 13.5.

I also have a mac Mini with M1 chip. I have not upgraded Homebrew on this mac and it is running 

```
> mosh-server --version
mosh-server (mosh 1.4.0) [build mosh-1.3.2.95rc1-11-gbc73a26-dirty]
```
I can connect to the m1 mini using Blink on the iPad.  I can also connect from the mini to the MacBook using mosh on the mini, so I know that the mosh-server version on the MacBook is working properly. But it doesn't work any longer from the iPad.

Any suggestions?

**Comment by tthkbw** [comment]
Well, I guess my searching wasn't so good. In perusing issue #1801 I played with alternate host address formulations. If I use the fixed ip4 address for my host, Blink and mosh now connect to the MacBook. However, if I use "MacAir.Local", which was in the host configuration and worked for the previous version of macOS (16.4) and mosh-server ([build mosh-1.3.2.95rc1-11-gbc73a26-dirty]), it does not work. 

So, what I know for macOS is this:

Using Homebrew mosh-server version "[build mosh-1.3.2.95rc1-11-gbc73a26-dirty]" works with "MacAir.Local" or with the IPv4 ip host address.

Using Homebrew mosh-server version "build mosh-1.4.0-3-gc16108f-Homebrew" does not work with host address "MacAir.Local", but does work with host address IPv4 ip address.

**Reply by carloscabanero** [reply, marked as answer]
We just pushed a fix for this on 16.1.12. Should hopefully be approved soon.
